repository_service_tuf_worker/services/keyvault/local.py

```diff
@@ -17,103 +21,192 @@
-                secrets_password = f.read().rstrip("\n")
+    @classmethod
+    def _base64_key(cls, keyvault_path, base64_key_body) -> str:
+        """
+        Decode a base64 key body and store in a file using unique hash file
+        name (based in the data) in the `keyvault_path`
+        (`RSTUF_LOCAL_KEYVAYLT_PATH`) and return the key filename.
+        Args:
+            keyvault_path: The key vault path defined in
+                `RSTUF_LOCAL_KEYVAYLT_PATH`.
+            base64_key_body: The key body on base64
+        Returns: key file_name as str
+        """
+        key_content = base64_key_body
+        hash_key = hashlib.blake2b(key_content.encode("utf-8"), digest_size=16)
+        file_name = hash_key.hexdigest()
+        key_path = os.path.join(keyvault_path, file_name)
+        if os.path.isfile(key_path):
+            return file_name
+        else:
+            with open(key_path, "w") as f:
+                f.write(base64.b64decode(key_content).decode())
+
+            return file_name
 
-            self._password = secrets_password
+    @classmethod
+    def _raw_key_parser(cls, path: str, keys: str) -> List[LocalKey]:
+        """
+        Parses the key(s) given in the `RSTUF_LOCAL_KEYVAULT_KEYS` and returns
+        as `LocalKey` object.
+        """
+        parsed_keys: List[LocalKey] = []
+        for raw_key in keys.split(":"):
+            if raw_key.startswith("/run/secrets/"):
+                # The user has stored their keys using container secrets.
+                with open(raw_key) as f:
+                    key_data = f.read().rstrip("\n").split(",")
+            else:
+                key_data = raw_key.split(",")
+
+            if len(key_data) == 1:
+                logging.error("Key is invalid")
+                pass
+
+            if len(key_data) == 2:  # filename and password
+                file = key_data[0]
+                password = key_data[1]
+                if file.startswith("base64|"):
+                    file = cls._base64_key(path, file.split("base64|")[1])
+                parsed_keys.append(LocalKey(file=file, password=password))
+
+            elif len(key_data) == 3:  # filename, type, password
+                file = key_data[0]
+                password = key_data[1]
+                key_type = key_data[2]
+                if file.startswith("base64|"):
+                    file = cls._base64_key(path, file.split("base64|")[1])
+                parsed_keys.append(
+                    LocalKey(file=file, password=password, type=key_type)
+                )
+            else:
+                logging.error("Key is invalid")
+                pass
 
-        self._secrets_handler: Callable = lambda *a: self._password
+        if len(parsed_keys) == 0:
+            raise KeyVaultError(
+                "No valid keys in configuration 'RSTUF_LOCAL_KEYVAULT_KEYS'"
+            )
+
+        return parsed_keys
 
     @classmethod
     def configure(cls, settings) -> None:
```

Comment:
Let's make `configure` the main entry point to create `LocaKeyVault` objects and validate them.